lib/ansible/modules/commands/command.py

```diff
@@ -213,7 +213,8 @@ def main():
     if not shell and args:
         args = shlex.split(args)
 
-    args = args or argv
+    # All args must be strings
+    args = [str(arg) for arg in (args or argv)]
```

Comment:
Do this in two statements instead of one for clarity.

also use to_native instead of str for safety:

``` python
args = args or argv
args = [to_native(arg, errors='surrogate_or_strict', nonstring='simplerepr') for arg in args]
```

str can traceback with non-ascii characters on Python2.  to_native will use utf-8 as it's default encoding so it's what we should choose.

If you care about details, run_command always uses bytes internally so you could use to_bytes here to transform into bytes instead of native strings.  However, the code uses args verbatim in some of the ```exit_json()``` calls below and those would have to use text on Python3 (and should use text on Python2) so native strings are the best choice.